File: db_connect.py
import mysql.connector
from mysql.connector import Error
import os
from dotenv import load_dotenv

# Load environment variables from .env file
load_dotenv()

def get_db_connection():
    """
    Establish and return a MySQL database connection.
    
    Returns:
        mysql.connector.connection.MySQLConnection: Database connection object if successful
        None: If connection fails
    """
    try:
        connection = mysql.connector.connect(
            host=os.getenv('DB_HOST'),
            user=os.getenv('DB_USER'),
            password=os.getenv('DB_PASSWORD'),
            database=os.getenv('DB_NAME')
        )
        
        if connection.is_connected():
            print("Successfully connected to MySQL Server")
            return connection
            
    except Error as e:
        print(f"Error: Could not connect to MySQL Server - {e}")
        return None

File: db_connect.js
const mysql = require('mysql2/promise');
require('dotenv').config();

async function getDbConnection() {
    try {
        const connection = await mysql.createConnection({
            host: process.env.DB_HOST,
            user: process.env.DB_USER,
            password: process.env.DB_PASSWORD,
            database: process.env.DB_NAME
        });

        return connection;
    } catch (error) {
        console.error('Error connecting to MySQL Server:', error.message);
        return null;
    }
}

module.exports = { getDbConnection };